lib/mongo/uri.rb

```diff
@@ -426,29 +426,38 @@ def self.uri_option(uri_key, name, extra = {})
     uri_option 'replicaset', :replica_set, :type => :replica_set
 
     # Timeout Options
-    uri_option 'connecttimeoutms', :connect_timeout, :type => :ms_convert
-    uri_option 'sockettimeoutms', :socket_timeout, :type => :ms_convert
-    uri_option 'serverselectiontimeoutms', :server_selection_timeout, :type => :ms_convert
-    uri_option 'localthresholdms', :local_threshold, :type => :ms_convert
+    uri_option 'connecttimeoutms', :connect_timeout, :type => :connect_timeout
+    uri_option 'sockettimeoutms', :socket_timeout, :type => :socket_timeout
+    uri_option 'serverselectiontimeoutms', :server_selection_timeout, :type => :server_selection_timeout
+    uri_option 'localthresholdms', :local_threshold, :type => :local_threshold
+    uri_option 'heartbeatfrequencyms', :heartbeat_frequency, :type => :heartbeat_frequency
+    uri_option 'maxidletimems', :max_idle_time, :type => :max_idle_time
 
     # Write Options
     uri_option 'w', :w, :group => :write
-    uri_option 'journal', :j, :group => :write
+    uri_option 'journal', :j, :group => :write, :type => :journal
     uri_option 'fsync', :fsync, :group => :write
-    uri_option 'wtimeoutms', :timeout, :group => :write
+    uri_option 'wtimeoutms', :timeout, :group => :write, :type => :wtimeout
 
     # Read Options
     uri_option 'readpreference', :mode, :group => :read, :type => :read_mode
     uri_option 'readpreferencetags', :tag_sets, :group => :read, :type => :read_tags
-    uri_option 'maxstalenessseconds', :max_staleness, :group => :read
+    uri_option 'maxstalenessseconds', :max_staleness, :group => :read, :type => :max_staleness
 
     # Pool options
     uri_option 'minpoolsize', :min_pool_size
     uri_option 'maxpoolsize', :max_pool_size
-    uri_option 'waitqueuetimeoutms', :wait_queue_timeout, :type => :ms_convert
+    uri_option 'waitqueuetimeoutms', :wait_queue_timeout, :type => :wait_queue_timeout
 
     # Security Options
     uri_option 'ssl', :ssl
+    uri_option 'tls', :ssl
+    uri_option 'tlsallowinvalidcertificates', :ssl_verify, :type => :ssl_verify
+    uri_option 'tlscafile', :ssl_ca_cert
+    uri_option 'tlsclientcertfile', :ssl_cert
+    uri_option 'tlsclientkeyfile', :ssl_key
+    uri_option 'tlsclientkeypassword', :ssl_key_pass_phrase
+    uri_option 'tlsinsecure', :ssl_verify, :type => :ssl_verify
```

Comment:
The documentation states that ssl_verify means:

> Whether to perform peer certificate validation.

Either the documentation needs to be updated to reflect the new behavior, which to me seems like the better option, or `tlsInsecure` cannot be an inverse of `ssl_verify`.

Reply:
Fixed in documentation